### PR TITLE
coreos/{kernel,sources}: bump to 4.0

### PR DESCRIPTION
**Pull request by mischief**

### CODE REVIEW

**Comment by gregkh** [comment]
lgtm

**Comment by marineam** [comment]
I would like to hold off until next week to merge this. Plenty of other stuff already going into this week's alpha and I like give major releases a little bit of bake time in case any last-minute bugs pop up in the community.

**Comment by marineam** [comment]
or, more specifically, merge after this week's alpha is tagged. @crawford ?

**Comment by crawford** [comment]
@marineam +1. I'll let you know once I'm ready.

File: sys-kernel/coreos-kernel/files/amd64_defconfig-4.0 [hunk 1]
```diff
@@ -1965,6 +1974,7 @@ CONFIG_MOUSE_PS2=m
 # CONFIG_MOUSE_PS2_ELANTECH is not set
 # CONFIG_MOUSE_PS2_SENTELIC is not set
 # CONFIG_MOUSE_PS2_TOUCHKIT is not set
+CONFIG_MOUSE_PS2_FOCALTECH=y
```
Comment (marineam) [inline sys-kernel/coreos-kernel/files/amd64_defconfig-4.0]: Is this pulled in automatically or is it optional? We don't have any other special mice

Comment (mischief) [inline sys-kernel/coreos-kernel/files/amd64_defconfig-4.0]: automatically.. i can remove it.

**Comment by mischief** [comment]
@marineam this was made with a `make olddefconfig` in a v4.0 git checkout. is that the correct way to update the kernel conf for coreos?

**Comment by marineam** [comment]
@mischief I generally do oldconfig so I can review new/changed options. Most of the time the defaults are fine but not everything is applicable to CoreOS

File: sys-kernel/coreos-kernel/files/amd64_defconfig-4.0 [hunk 2]
```diff
@@ -1965,6 +1974,7 @@ CONFIG_MOUSE_PS2=m
 # CONFIG_MOUSE_PS2_ELANTECH is not set
 # CONFIG_MOUSE_PS2_SENTELIC is not set
 # CONFIG_MOUSE_PS2_TOUCHKIT is not set
+# CONFIG_MOUSE_PS2_FOCALTECH=y
```
Comment (marineam) [inline sys-kernel/coreos-kernel/files/amd64_defconfig-4.0]: You didn't re-run this through oldconfig did you :-P

**Comment by marineam** [comment]
amd64_defconfig-3.19 was removed in previous diffs, now it is not

**Comment by marineam** [comment]
[LGTM](http://www.lgtm.in/i/IWib6R02q)

**Pull request merged**